pydoctor/astbuilder.py

```diff
@@ -512,36 +485,81 @@ def _annotation_from_attrib(self, expr, ctx):
                 return _infer_type(default)
         return None
 
-    def _unstring_annotation(self, node):
+    def _annotations_from_function(self, func: ast.FunctionDef) -> Mapping[str, ast.expr]:
+        """Get annotations from a function definition.
+        @param func: The function definition's AST.
+        @return: Mapping from argument name to annotation.
+            The name C{return} is used for the return type.
+            Unannotated arguments are omitted.
+        """
+        def _get_all_args() -> Iterator[ast.arg]:
+            base_args = func.args
+            # New on Python 3.8 -- handle absence gracefully
+            try:
+                yield from base_args.posonlyargs
+            except AttributeError:
+                pass
+            yield from base_args.args
+            varargs = base_args.vararg
+            if varargs:
+                yield varargs
+            yield from base_args.kwonlyargs
+            kwargs = base_args.kwarg
+            if kwargs:
+                yield kwargs
+        def _get_all_ast_annotations() -> Iterator[Tuple[str, ast.expr]]:
+            for arg in _get_all_args():
+                if arg.annotation:
+                    yield arg.arg, arg.annotation
+            returns = func.returns
+            if returns:
+                yield 'return', returns
+        return {name: self._unstring_annotation(value)
```

Comment:
Couldn't this just have been

```
return {name: self._unstring_annotation(value) for name, value in get_function_signature_annotation(func).items()
```
And not required moving the whole function in here?

Behavioral change diffs (even regression fix ones) are easier to review when they're separated from a zero-behavior-change refactoring review.

Reply:
Without making the function into a method, there is no `self`, so I don't see how that would work.

I agree with separating the changes, but that's why I put the move in a commit by itself. I'm not sure what more I could have done.